test/test_utils.cpp

```diff
@@ -386,3 +387,67 @@ TEST(ModelReading, equations)
         EXPECT_DOUBLE_EQ(node.x(), node.y());
 }
 
+TEST(Quaternion, addition)
+{
+    biorbd::utils::Quaternion q1(1,2,3,4);
+    biorbd::utils::Quaternion q2(2,3,4,5);
+    EXPECT_NEAR(q1[3], 1, requiredPrecision);
+    EXPECT_NEAR(q1[0], 2, requiredPrecision);
+    EXPECT_NEAR(q1[1], 3, requiredPrecision);
+    EXPECT_NEAR(q1[2], 4, requiredPrecision);
+    EXPECT_NEAR(q2[3], 2, requiredPrecision);
+    EXPECT_NEAR(q2[0], 3, requiredPrecision);
+    EXPECT_NEAR(q2[1], 4, requiredPrecision);
+    EXPECT_NEAR(q2[2], 5, requiredPrecision);
+    biorbd::utils::Quaternion q12 = q1+q2;
+    EXPECT_NEAR(q12[3], 3, requiredPrecision);
+    EXPECT_NEAR(q12[0], 5, requiredPrecision);
+    EXPECT_NEAR(q12[1], 7, requiredPrecision);
+    EXPECT_NEAR(q12[2], 9, requiredPrecision);
+}
+
+
+TEST(Quaternion, multiplication)
+{
+    biorbd::utils::Quaternion q1(1,2,3,4);
+    biorbd::utils::Quaternion q2(2,3,4,5);
+    EXPECT_NEAR(q1[3], 1, requiredPrecision);
+    EXPECT_NEAR(q1[0], 2, requiredPrecision);
+    EXPECT_NEAR(q1[1], 3, requiredPrecision);
+    EXPECT_NEAR(q1[2], 4, requiredPrecision);
+    EXPECT_NEAR(q2[3], 2, requiredPrecision);
+    EXPECT_NEAR(q2[0], 3, requiredPrecision);
+    EXPECT_NEAR(q2[1], 4, requiredPrecision);
+    EXPECT_NEAR(q2[2], 5, requiredPrecision);
+    biorbd::utils::Quaternion q12 = q1*q2;
+    EXPECT_NEAR(q12[3], -36, requiredPrecision);
+    EXPECT_NEAR(q12[0], 6, requiredPrecision);
+    EXPECT_NEAR(q12[1], 12, requiredPrecision);
+    EXPECT_NEAR(q12[2], 12, requiredPrecision);
+}
+
+TEST(Quaternion, derivative)
+{
+    biorbd::utils::Quaternion q1;
```

Comment:
Again, is there a risk of false positive when using the unitary quaternion?

Reply:
Idem